src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const [user, setUser] = useState<User | null>(null);
+
+  if (!currentTodo) {
+    throw Error('Not todo');
+  }
+
+  useEffect(() => {
+    getUser(currentTodo.userId).then(userFromServer => setUser(userFromServer));
+  }, [currentTodo]);
+
+  const handleCloseModal = () => {
+    dispatch(currentTodoSlice.actions.set(null));
```

Comment:
Ensure that the `set` action is correctly defined in the `currentTodoSlice`. If this action does not exist, it will cause an error when trying to dispatch it.